packages/data/src/plugins/async-generator/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { applyMiddleware } from 'redux';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import asyncGeneratorMiddleware, { toAsyncIterable } from '../../middlewares/async-generator';
+
+export default function( registry ) {
+	return {
+		registerStore( reducerKey, options ) {
+			const store = registry.registerStore( reducerKey, options );
+			const enhancer = applyMiddleware( asyncGeneratorMiddleware );
+			const createStore = () => store;
+			Object.assign(
+				store,
+				enhancer( createStore )( options.reducer )
+			);
+			return store;
+		},
+
+		async fulfill( reducerKey, selectorName, ...args ) {
+			const resolver = get( registry.namespaces, [ reducerKey, 'resolvers', selectorName ] );
+			if ( ! resolver ) {
+				return;
+			}
+			const store = registry.namespaces[ reducerKey ].store;
+			const state = store.getState();
+			const actionCreator = resolver.fulfill( state, ...args );
+			if ( actionCreator ) {
+				await store.dispatch( toAsyncIterable( actionCreator ) );
```

Comment:
So the idea with the stock middlewares (promise, async generator) is that they're expected to be present for use by plugins? Would it not be possible for the plugins themselves to ensure their presence if needed? Via middleware application in the overrided `registerStore` ?

Reply:
Isn't it the case already? 

Aside the `promise` middleware which is considered the "core" implementation of the resolvers and that's why it's not in a plugin. The others are already.